### DISCUSSION TITLE
The locations or offsets of sentences in a text

### DISCUSSION THREAD

**Discussion by hehuan2112** [nlp-compromise/nlp_compromise#892] [Q&A] [answered]
Hi there,

This may be another silly question but how to get the offsets of sentences in a text?
For example, I'd like to get the sentences from the below text:

```
It is a fox. It is a fox. Is that a fox?
```

With the current code, I can get the three sentences:

```
nlp(text).sentences().json()
(3) [{…}, {…}, {…}]
0: {text: 'It is a fox.', terms: Array(4)}
1: {text: 'It is a fox.', terms: Array(4)}
2: {text: 'Is that a fox?', terms: Array(4)}

```

And in my case, I need to get the offsets of each sentence in original text. For example:

```
0: {text: 'It is a fox.', start:0, end: 12, terms: Array(4)}
1: {text: 'It is a fox.', start: 13, end: 25, terms: Array(4)}
2: {text: 'Is that a fox?', start: 26, end: 40, terms: Array(4)}
```

Currently I am using `text.indexOf(sentence)` to get the offset.
So, sometime when there are duplicated sentences, it need to check again.
I'm wondering if there is any internal parameter or function I could use to get those offsets

**Comment by spencermountain** [comment, marked as answer]
hey Huan - yup you can do that like this:
```js
doc.sentences().json({offset:true})
```
https://runkit.com/spencermountain/61bc982d7cf0930008a200a4
sorry it's not very clear in the docs right now
cheers

**Reply by hehuan2112** [reply]
Wow! It works! That's super helpful for me!!!
Thank you so much for the timely answer :)
Have a great weekend and holiday!

Huan

**Reply by ryanj11** [reply]
This helped me too. Thanks!
